demo/switch.js

```diff
@@ -17,18 +18,23 @@ var img_map;
 var base_title;
 function sw_init(m) {
   img_map = m + '.png';
-  switch (m) {
-    case "map_general": base_title = "Bendras"; break;
-    case "map": base_title = "Lankytinos"; break;
-    case "map_bicycle": base_title = "Dviračiai"; break;
-    case "map_topo": base_title = "Topografinis"; break;
-    case "map_upes": base_title = "Upės (baidarės)"; break;
-  }
   c = document.getElementById('sw_container');
   c.classList.add('sw_container');
   c.innerHTML = '<div id="sw_extend" onClick="sw_toggle()"><img id="sw_button" src="expand_button.png"></div><div id="sw_list" class="sw_list"></div>';
   e = document.getElementById('sw_list');
   e.innerHTML = s_initial;
+  switch (m) {
+    case "map_general": base_title = "Bendras"; break;
+    case "map":         base_title = "Lankytinos"; break;
+    case "map_bicycle": base_title = "Dviračiai"; break;
+    case "map_topo":    base_title = "Topografinis"; break;
+    case "map_upes":    base_title = "Upės (baidarės)"; break;
+  }
+  if (m != 'map_general') { s_extended += s_ex1; }
+  if (m != 'map')         { s_extended += s_ex2; }
+  if (m != 'map_bicycle') { s_extended += s_ex3; }
+  if (m != 'map_topo')    { s_extended += s_ex4; }
+  if (m != 'map_upes')    { s_extended += s_ex5; }
```

Comment:
Siūlyčiau pagražinti šią vietą, panaudojant masyvą.
Parašysiu fragmenta kaip būtu galima pasidaryti objektą, poto iš to objekto išmesti ko nereikia ir padaryti tavo norimą stringą. Tikiuosi suprasi, jei ką rašyk:
```
var s_ex = {
  map_general: '<div id="sw1" class="sw_map sw_map1 sw_map_hidden" onClick="sw_switch_to_map(\'-\')"><img src="map_general.png"><br>Bendras</div>',
  map: '<div id="sw2" class="sw_map sw_map2 sw_map_hidden" onClick="sw_switch_to_map(\'places\')"><img src="map.png"><br>Lankytinos</div>',
  map_bicycle: '<div id="sw3" class="sw_map sw_map3 sw_map_hidden" onClick="sw_switch_to_map(\'dviraciai\')"><img src="map_bicycle.png"><br>Dviračiai</div>',
  map_topo: '<div id="sw4" class="sw_map sw_map4 sw_map_hidden" onClick="sw_switch_to_map(\'topo\')"><img src="map_topo.png"><br>Topografinis</div>',
  map_upes: '<div id="sw5" class="sw_map sw_map5 sw_map_hidden" onClick="sw_switch_to_map(\'upes\')"><img src="map_upes.png"><br>Upės (baidarės)</div>'
}
var s_extended = '';
var base_title;

function sw_init(m) {
  ...
  switch (m) {
    case "map_general": base_title = "Bendras"; break;
    case "map":         base_title = "Lankytinos"; break;
    case "map_bicycle": base_title = "Dviračiai"; break;
    case "map_topo":    base_title = "Topografinis"; break;
    case "map_upes":    base_title = "Upės (baidarės)"; break;
  }
  var s_extended = Object.keys(s_ex)
  .reduce(function(prev, key) {
    return prev + (key === m ? '' : s_ex[key]);
  }, '');
}

sw_init('map_bicycle');
```
Buvo panaudotos funkcijos https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys ir https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce

Reply:
Dėl `base_title` būtu galimybė pasidaryti tokį taip:
```
var s_ex = {
  map_general: {
    title: 'Bendras',
    html: '<div id="sw1" class="sw_map sw_map1 sw_map_hidden" onClick="sw_switch_to_map(\'-\')"><img src="map_general.png"><br>Bendras</div>'
  },
  map: {
    title: 'Lankytinos',
    html: '<div id="sw2" class="sw_map sw_map2 sw_map_hidden" onClick="sw_switch_to_map(\'places\')"><img src="map.png"><br>Lankytinos</div>'
  },
  map_bicycle: {
    title: 'Dviračiai',
    html: '<div id="sw3" class="sw_map sw_map3 sw_map_hidden" onClick="sw_switch_to_map(\'dviraciai\')"><img src="map_bicycle.png"><br>Dviračiai</div>'
  },
  map_topo: {
    title: 'Topografinis',
    html: '<div id="sw4" class="sw_map sw_map4 sw_map_hidden" onClick="sw_switch_to_map(\'topo\')"><img src="map_topo.png"><br>Topografinis</div>'
  },
  map_upes: {
    title: 'Upės (baidarės)',
    html: '<div id="sw5" class="sw_map sw_map5 sw_map_hidden" onClick="sw_switch_to_map(\'upes\')"><img src="map_upes.png"><br>Upės (baidarės)</div>'
  }
}
var s_extended = '';
var base_title = '';

function sw_init(m) {
  ...
  base_title = s_ex[m] ? s_ex[m].title : '';
  s_extended = Object.keys(s_ex)
  .reduce(function(prev, key) {
    return prev + (key === m ? '' : s_ex[key].html);
  }, '');
}

sw_init('map_bicycle');
```